### PR TITLE
Introduce `*.aab` support

### PR DESCRIPTION
**Pull request by mateherber**
### Checklist

- [✔️] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MAJOR_ [version update](https://semver.org/)

### Context

Google is making `*.aab` bundles a mandatory format to publish updates in the play store
https://android-developers.googleblog.com/2021/06/the-future-of-android-app-bundles-is.html

This change introduces `*.aab` support to the step.

Resolves: #70

### Changes

- Updated dependencies
- Introduced `android_build_type` input to control if `*.apk` or `*.aab` is the expected output for android builds
- Migrated to new CI system

### Decisions

We decided to introduce a flag that directly controls the package type similar to the android build steps instead of relying on the generic options input field.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -228,10 +234,22 @@ func main() {
...
 	}
 
+	log.Printf("cordova version: %s", colorstring.Green(cordovaVersion.String()))
+	
+	minCordovaVersion, err := ver.NewVersion("8.1.0")
+	if err != nil {
+		fail("Failed to parse version 8.1.0: %s", err)
+	}
+	if cordovaVersion.LessThan(minCordovaVersion) {
+		log.Warnf("Cordova doesn't support exporting aab, falling back to apk")
```
Comment (ofalvai) [inline main.go:249]: This is printed even if the config is set to APK (`isAAB` is false)

Comment (mateherber) [reply main.go:249]: True, thanks! Fixing

File: main.go [hunk 2]
```diff
@@ -432,35 +454,47 @@ func main() {
...
+			pathEnvKey := apkPathEnvKey
+			if isAAB {
+				pathEnvKey = aabPathEnvKey
+			}
+			pathListEnvKey := apkPathListEnvKey
+			if isAAB {
+				pathListEnvKey = aabPathListEnvKey
+			}
+			if exportedPth, exportedPaths, err := moveAndExportOutputs(distPkg, configs.DeployDir, pathEnvKey, pathListEnvKey); err != nil {
 				fail("Failed to export apks, error: %s", err)
```
Comment (ofalvai) [inline main.go:485]: Outdated error message

File: main.go [hunk 3]
```diff
@@ -432,35 +454,47 @@ func main() {
...
 			} else if exportedPth != "" {
-				log.Donef("The apk path is now available in the Environment Variable: %s (value: %s)", apkPathEnvKey, exportedPth)
+				log.Donef("The apk path is now available in the Environment Variable: %s (value: %s)", pathEnvKey, exportedPth)
```
Comment (ofalvai) [inline main.go:487]: This is not always the APK path despite first part of the message

File: step.yml [hunk 4]
```diff
@@ -27,18 +27,18 @@ description: |-
...
-  
+
 website: https://github.com/bitrise-steplib/steps-ionic-archive
 source_code_url: https://github.com/bitrise-steplib/steps-ionic-archive
 support_url: https://github.com/bitrise-steplib/steps-ionic-archive/issues
 project_type_tags:
-- ionic
+  - ionic
 type_tags:
-- build
```
Comment (ofalvai) [inline step.yml:37]: I think these are unintended changes (our yml standard is to start the list with the same indentation as the parent, right?)

Comment (mateherber) [reply step.yml:37]: I'm using GoLand default YML indentation settings. Is there a place where we defined how do we indent yml? 🤔

Comment (mateherber) [reply step.yml:37]: In the devcenter samples, they are not indented indeed.

**Review by ofalvai — APPROVED** [review]

**Review by ofalvai — APPROVED** [review]

**Pull request merged**